Compose and datadir restructure

maurochss opened the pull request.

maurochss committed:
Deleted

maurochss committed:
Moved to mysql/conf.d

maurochss committed:
Removed new empty folder approach

maurochss committed:
Removed REDME.md and implement .gitkeep

maurochss committed:
Added exclusions for database data and logs

maurochss committed:
Moved from mysql into mysql/conf.d/mysql.conf

maurochss committed:
Change mysql.conf mapping, change database host mapping for logs and data location into /var/lib/sesp/mysql and /var/log/sesp/mysql

Guimino commented on docker-compose.yml:
```diff
@@ -19,11 +19,10 @@ services:
     volumes:
       - /etc/localtime:/etc/localtime
-      - ./database/conf/mysql.cnf:/etc/mysql/mysql.conf
-      - ./database/logs:/var/log/mysql
       - ./database/conf:/etc/mysql
-      - ./database/data:/var/lib/mysql
```
@maurochss, para o base de dados peco alterares para volume gerenciado pelo Docker
 **- refapp-data:/var/lib/mysql**
 
 Sendo possivel, podemos deixar os logs na forma Bind mounts
 **/var/log/sesp/mysql:/var/log/mysql**

Guimino replied:
@maurochss  aproveitando, acho que podemos usar o arquivo env para armazenar as variaves de ambiente.
Haveria de solicitar para reveres a seccao das variaveis de ambiente do compose, algo como :
environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
      - MYSQL_USER=${MYSQL_USER}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
      
     environment:
      DISA_ENVIRONMENT: ${DISA_ENVIRONMENT:-production}
      DISA_API_USERNAME: ${DISA_API_USERNAME}
      DISA_API_PASSWORD: ${DISA_API_PASSWORD}
 
 E o sample **env** poderia ter os pares:
MYSQL_ROOT_PASSWORD=root
MYSQL_DATABASE=openmrs_prod
MYSQL_USER=sesp
MYSQL_PASSWORD=sesp
DISA_ENVIRONMENT=production
DISA_API_USERNAME=disa_api_user
DISA_API_PASSWORD=api_api_password

podemos comitar o env na raiz do projecto e actualizamos o Read.me acrescentando a instrucao : `mv env .env`

maurochss committed:
Implementation of Docker volumes

maurochss committed:
Changed mysql.conf to mysql.cnf

maurochss committed:
ignore tomcat logs

maurochss committed:
Exclude tomcat logs

maurochss committed:
1. Add volume approach using refappDb
2. Add MYSQL_LOG_DIRECTORY
3. Fix space after stop grace period and timeout
4. Unify Database volume for logs and data

maurochss committed:
1. Rescructure directory validation, since we now have data and log directories;
2. Set file and directory owner to mysql.
3. Get temporary password from mysql_error.log instead of mysql_startup_log.txt
4. Remove forcing of sql_mode since this is handled by mysql.cnf

maurochss committed:
1. Change max_allowed_packet to the mysql official max value.
2. Add line feed at the end.
